File: app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsPresenterImpl.java
package com.esoxjem.movieguide.details;

import com.esoxjem.movieguide.Movie;

/**
 * @author arun
 */
public class MovieDetailsPresenterImpl implements MovieDetailsPresenter {
    private MovieDetailsView view;

    public MovieDetailsPresenterImpl() {
    }

    @Override
    public void setView(MovieDetailsView view) {
        this.view = view;
    }

    @Override
    public void destroy() {
        view = null;
    }

    @Override
    public void showDetails(Movie movie) {
        if (isViewAttached()) {
            view.showDetails(movie);
        }
    }

    private boolean isViewAttached() {
        return view != null;
    }
}

File: app/src/main/java/com/esoxjem/movieguide/details/MovieDetailsPresenterImpl.kt
package com.esoxjem.movieguide.details

import com.esoxjem.movieguide.Movie

/**
 * @author arun
 */
class MovieDetailsPresenterImpl : MovieDetailsPresenter {

    private var view: MovieDetailsView? = null

    private val isViewAttached: Boolean
        get() = view != null

    override fun setView(view: MovieDetailsView) {
        this.view = view
    }

    override fun destroy() {
        view = null
    }

    override fun showDetails(movie: Movie) {
        if (isViewAttached) {
            view?.showDetails(movie)
        }
    }
}
